device/iot-device-samples/device-reconnection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceClientManager.java

```diff
@@ -0,0 +1,72 @@
+package samples.com.microsoft.azure.sdk.iot;
+
+import com.microsoft.azure.sdk.iot.device.DeviceClient;
+import lombok.experimental.Delegate;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+
+@Slf4j
+public class DeviceClientManager {
+    private static final Object lock = new Object();
+    private static boolean connecting;
+    @Delegate
+    private DeviceClient client;
+
+    DeviceClientManager(DeviceClient deviceClient) {
+        this.client = deviceClient;
+        this.client.registerConnectionStatusChangeCallback(new IotHubConnectionStatusChangeCallbackHandler(), this);
+    }
+
+    void connect() {
+        // Device client does not have retry on the initial open() call. Will need to be re-opened by the calling application
+        while (true) {
+            try {
+                log.debug("[connect] - Opening the device client instance...");
+                client.open();
+                break;
+            }
+            catch (Exception ex) {
+                log.error("[connect] - Exception thrown while opening DeviceClient instance.");
+                ex.printStackTrace();
+            }
+            try {
+                log.debug("[connect] - Sleeping for 10 secs before attempting another open()");
+                Thread.sleep(10 * 1000);
+            }
```

Comment:
Do you need this delay?

Reply:
No, this delay is not needed for the device client to function; it has simply been added so that we do not cause any load spike when we try to keep on opening immediately after failing (in the event of network loss etc).